pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -126,6 +127,18 @@ def solve_master(solve_data, config, feas_pump=False):
         masteropt.options['timelimit'] = config.time_limit
     if config.threads > 0:
         masteropt.options["threads"] = config.threads
+    if config.use_tabu_list:
+        tabu_list = masteropt._solver_model.register_callback(
+            IncumbentCallback_cplex)
+        tabu_list.solve_data = solve_data
+        tabu_list.opt = masteropt
+        masteropt._solver_model.parameters.preprocessing.reduce.set(1)
+        # If the callback is used to reject incumbents, the user must set the 
+        # parameter c.parameters.preprocessing.reduce either to the value 1 (one) 
+        # to restrict presolve to primal reductions only or to 0 (zero) to disable all presolve reductions
```

Comment:
Does this mean that those extra parameters should be fixed by us or rely on the final user to do it? Is this note for us the developers?

Reply:
This is the note for us and it's from the CPLEX website. I test it.
In cplex, if the incumbent solution (optimal) is found at the root node(before branch & bound), the incumbent callback will not be called. This often happens in small MIPs.
If we set this parameter to 1, it will prevent the above situation from happening.

Reply:
This was the same note in the paper. I think they handled it by actually adding the no-good cut in that case. I don't want to turn off the presolve.

Reply:
Actually, I am not turning off the whole presolve. I just turn off dual reduction and the primal reduction still applies. 

Incumbent callback reference:
https://www.ibm.com/support/knowledgecenter/en/SSSA5P_12.7.1/ilog.odms.cplex.help/refpythoncplex/html/cplex.callbacks.IncumbentCallback-class.html

Preprocessing reference:
https://www.ibm.com/support/knowledgecenter/SSSA5P_12.9.0/ilog.odms.cplex.help/CPLEX/Parameters/topics/Reduce.html

Reply:
As you can see in the document, the dual reduction will be automatically turned off if `lazyconstraint` callback is included.